File: plugin-rust-agent/src/main/java/jetbrains/buildServer/rust/cargo/YankArgumentsProvider.java
/*
 * Copyright 2000-2016 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * See LICENSE in the project root for license information.
 */

package jetbrains.buildServer.rust.cargo;

import jetbrains.buildServer.rust.ArgumentsProvider;
import jetbrains.buildServer.rust.CargoConstants;
import jetbrains.buildServer.util.StringUtil;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * Provides arguments to cargo yank command.
 */
public class YankArgumentsProvider implements ArgumentsProvider {

    @NotNull
    @Override
    public List<String> getArguments(@NotNull final Map<String, String> parameters) {
        final List<String> arguments = new ArrayList<String>();
        arguments.add(CargoConstants.COMMAND_YANK);

        final String versionValue = parameters.get(CargoConstants.PARAM_YANK_VERSION);
        if (!StringUtil.isEmptyOrSpaces(versionValue)) {
            arguments.add("--vers");
            arguments.add(versionValue.trim());
        }

        final String undoValue = parameters.get(CargoConstants.PARAM_YANK_UNDO);
        if ("true".equalsIgnoreCase(undoValue)) {
            arguments.add("--undo");
        }

        final String indexValue = parameters.get(CargoConstants.PARAM_YANK_INDEX);
        if (!StringUtil.isEmptyOrSpaces(indexValue)) {
            arguments.add("--index");
            arguments.add(indexValue.trim());
        }

        final String tokenValue = parameters.get(CargoConstants.PARAM_YANK_TOKEN);
        if (!StringUtil.isEmptyOrSpaces(tokenValue)) {
            arguments.add("--token");
            arguments.add(tokenValue.trim());
        }

        final String verbosityValue = parameters.get(CargoConstants.PARAM_VERBOSITY);
        if (!StringUtil.isEmptyOrSpaces(verbosityValue)) {
            arguments.add(verbosityValue.trim());
        }

        final String crateValue = parameters.get(CargoConstants.PARAM_YANK_CRATE);
        if (!StringUtil.isEmptyOrSpaces(crateValue)) {
            arguments.add(crateValue.trim());
        }

        return arguments;
    }
}

File: plugin-rust-agent/src/main/kotlin/jetbrains/buildServer/rust/cargo/YankArgumentsProvider.kt


package jetbrains.buildServer.rust.cargo

import jetbrains.buildServer.agent.BuildRunnerContext
import jetbrains.buildServer.rust.ArgumentsProvider
import jetbrains.buildServer.rust.CargoConstants
import java.util.*

/**
 * Provides arguments to cargo yank command.
 */
class YankArgumentsProvider : ArgumentsProvider {

    override fun getArguments(runnerContext: BuildRunnerContext): List<String> {
        val parameters = runnerContext.runnerParameters
        val arguments = ArrayList<String>()
        arguments.add(CargoConstants.COMMAND_YANK)

        val versionValue = parameters[CargoConstants.PARAM_YANK_VERSION]
        if (!versionValue.isNullOrBlank()) {
            arguments.add("--vers")
            arguments.add(versionValue.trim())
        }

        val undoValue = parameters[CargoConstants.PARAM_YANK_UNDO]
        if ("true".equals(undoValue, ignoreCase = true)) {
            arguments.add("--undo")
        }

        val indexValue = parameters[CargoConstants.PARAM_YANK_INDEX]
        if (!indexValue.isNullOrBlank()) {
            arguments.add("--index")
            arguments.add(indexValue.trim())
        }

        val tokenValue = parameters[CargoConstants.PARAM_YANK_TOKEN_SECURE] ?: parameters[CargoConstants.PARAM_YANK_TOKEN]
        if (!tokenValue.isNullOrBlank()) {
            arguments.add("--token")
            arguments.add(tokenValue.trim())
        }

        addCommonArguments(parameters, arguments)

        val crateValue = parameters[CargoConstants.PARAM_YANK_CRATE]
        if (!crateValue.isNullOrBlank()) {
            arguments.add(crateValue.trim())
        }

        return arguments
    }
}